ocaml/xapi-storage/python/xapi/storage/api/volume.py

```diff
@@ -1106,7 +1107,7 @@ def __init__(self):
     def probe(self, dbg, uri):
         """Operations which act on Storage Repositories"""
         result = {}
-        result["result"] = { "srs": [ { "sr": "string", "name": "string", "description": "string", "free_space": 0L, "total_space": 0L, "datasources": [ "string", "string" ], "clustered": True, "health": None }, { "sr": "string", "name": "string", "description": "string", "free_space": 0L, "total_space": 0L, "datasources": [ "string", "string" ], "clustered": True, "health": None } ], "uris": [ "string", "string" ] }
+        result["result"] = { "srs": [ { "sr": "string", "name": "string", "description": "string", "free_space": 0, "total_space": 0, "datasources": [ "string", "string" ], "clustered": True, "health": None }, { "sr": "string", "name": "string", "description": "string", "free_space": 0, "total_space": 0, "datasources": [ "string", "string" ], "clustered": True, "health": None } ], "uris": [ "string", "string" ] }
```

Comment:
What are the consequences of changing '0' to '0L' here? One will be 'int', the other will be 'long' in Python2. Can that potentially confuse the RPC serialization/deserialization glue code, will it handle both?

Reply:
In Python3, all integers are long.

Reply:
@edwintorok I pushed an update of this commit for fixing it to settle any doubts:

https://github.com/xapi-project/xen-api/pull/5078/commits/ac683ca9815f65ddc48bffdb4e242909387c3daa

Using `futurize --step1` would have replaced `0L` with `0`. But for Python2, that would have changed values from `long` to `int`. Fix this like `ocaml-rpc/src/lib/pythongen.ml` already does it for the generated code:

1. Add:
   ```py
   if sys.version_info[0] > 2:
       long = int
       unicode = str
       str = bytes
   ```

   This is already used in the code generated by by `pythongen.ml`:
   https://github.com/mirage/ocaml-rpc/blob/master/src/lib/pythongen.ml#L102

2. Apply:
   ```ml
   sed -i 's/ 0L, / long(0), /g' ocaml/xapi-storage/python/xapi/storage/api/volume.py
   ```
   This is already used in the code generated by by `pythongen.ml`:
   https://github.com/mirage/ocaml-rpc/blob/master/src/lib/pythongen.ml#L284